### PR TITLE
RTL: force (%) symbol to follow text direction

### PR DESCRIPTION
**Pull request by ARMBouhali**
**TL;DR -** This commit makes the percentage (%) symbol appear correctly after numbers for RTL languages.

**Background**
- In RTL languages such as Arabic, the correct placement for `%` is to come after the number, following the writing direction.
- For example, 25% in RTL should appear as %25 instead.

**What changed?**
- We use 2 frontend-platform functions `isRtl` and `getLocale` to apply the fix only if the current user language is right-to-left.
- To avoid using a space to enforce the `%` to follow the RTL direction, we use instead the [right-to-left marker character](https://en.wikipedia.org/wiki/Right-to-left_mark) which is a non-printing character that does the exact job.
- This change is applied to the following components in the progress tab:
   * Course completion donut chart: 'XX% complete' label
   * Grade Bar: current grade & passing grade tooltips.
   * Grade summary table

**Screenshots**
Faulty layout in RTL: the `%` sign appears before the value (to the right)
Screenshot from 2022-09-15 16-54-32

Fixed layout in RTL: the `%` sign appears correctly after the value (to the left)
Screenshot from 2022-09-15 16-52-22

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @ARMBouhali!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by natabene** [comment]
@ARMBouhali Thank you for the contribution, we will review this once your agreement has been sorted out.

**Comment by natabene** [comment]
@ARMBouhali Great, your CLA check is green, our team can review now.

**Comment by ARMBouhali** [comment]
I've updated the commit message to comply with `commitlint` rules.

File: src/course-home/progress-tab/grades/course-grade/CurrentGradeTooltip.jsx [hunk 1]
```diff
@@ -41,7 +41,7 @@ function CurrentGradeTooltip({ intl, tooltipClassName }) {
         overlay={(
           <Popover id={`${isPassing ? 'passing' : 'non-passing'}-grade-tooltip`} aria-hidden="true" className={tooltipClassName}>
             <Popover.Content data-testid="currentGradeTooltipContent" className={isPassing ? 'text-white' : 'text-dark-700'}>
-              {currentGrade.toFixed(0)}%
+              {currentGrade.toFixed(0)}{isLocaleRtl ? '\u200f' : ''}%
```
Comment (brian-smith-tcril) [inline src/course-home/progress-tab/grades/course-grade/CurrentGradeTooltip.jsx:44]: is there a specific reason to use the `_ ? _ : _` pattern here instead of hte `_ && _` pattern? it seems both patterns are used in this PR.

**Review by brian-smith-tcril — COMMENTED** [review]
it feels like there's a lot of redundant code in here, but nothing pops to mind as a better way to handle RTL percentages. @arbrandes any chance you have thoughts on any other ways to handle this?

File: src/course-home/progress-tab/grades/course-grade/CurrentGradeTooltip.jsx (hunk 1 above)
Comment (ARMBouhali) [reply src/course-home/progress-tab/grades/course-grade/CurrentGradeTooltip.jsx:44]: @brian-smith-tcril
> is there a specific reason to use the `_ ? _ : _` pattern here instead of hte `_ && _` pattern? it seems both patterns are used in this PR.

 indeed there is a reason; it was for a clean rendering for each context.
- `condition && text` was used to render a JSX element (DOM literal) as a shorthand for `condition ? text : null`.
- `condition ? text : ''` was needed to render a string within a string template, otherwise, when `condition` is false, `condition && text` would give a `'false'` string instead of an empty string `''`

> it feels like there's a lot of redundant code in here...

Concise ways to deal with this:
- Add RTL function(s) to  `frontend-platform/i18n` to do RTL operations. e.g `i18n.adjust('%')`
- Render a translated percentage expression using `intl.formatMessage` or `<FormattedMessage />` and let translators deal with each instance of `%` in every language as I did for the Arabic learning MFE strings. Transifex unfortunately does not show special chars so it's not possible to observe the result without exporting a translation file.

**Comment by mphilbrick211** [comment]
@mattcarter - Hi Mat! Could you please allow actions / tests to run on this item?

**Comment by natabene** [comment]
here you go

**Comment by ARMBouhali** [comment]
I've rebased the branch for Codecov tests to pass.

**Review by brian-smith-tcril — APPROVED** [review]
The ways to reduce redundancy mentioned in https://github.com/openedx/frontend-app-learning/pull/978#discussion_r993810989 feel out of scope for this PR (but it'd be great to make an issue for them)

I see no reason to wait for those changes before merging this.

LGTM!

**Comment by mphilbrick211** [comment]
> @mattcarter - Hi Mat! Could you please allow actions / tests to run on this item?

Friendly ping on this :)

**Comment by ARMBouhali** [comment]
@mattcarter @mphilbrick211 One last failing test, but it appears out of reach for me.
an error occurred with the coverage report upload:

    Error: Codecov: Failed to properly upload: The process '/home/runner/work/_actions/codecov/codecov-action/v3/dist/codecov' failed with exit code 255

Can anyone check what's actually wrong?

**Review by arbrandes — APPROVED** [review]
These changes look good to me, and tests are passing.  Approved.

Before merging, however, it would be good to have an ok from a frontend-app-learning dev.  If none emits an opinion in a day or so, I'll press the merge button.

**Comment by ARMBouhali** [comment]
@arbrandes FYI a [similar PR for frontend-app-gradebook](https://github.com/openedx/frontend-app-gradebook/pull/265) has been approved and merged a while ago.

**Comment by mphilbrick211** [comment]
> These changes look good to me, and tests are passing. Approved.
> 
> Before merging, however, it would be good to have an ok from a frontend-app-learning dev. If none emits an opinion in a day or so, I'll press the merge button.

@mattcarter

**Comment by openedx-webhooks** [comment]
@ARMBouhali 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-app-learning&entry.867055334=https://github.com/openedx/frontend-app-learning/pull/978&entry.1484655318=https://github.com/ARMBouhali&entry.752974735=2022-09-15+16:18&entry.1917517419=2022-10-26+13:31&entry.2133058324=Yes) so we can improve your experience in the future.
